### ISSUE TITLE
removeLayer method throws an error

### ISSUE THREAD

**Issue by ttsvetko** [ismyrnow/leaflet-groupedlayercontrol#49]
Here it the code-fragment for 'removeLayer':

```javascript
 removeLayer: function (layer) {
    var id = L.Util.stamp(layer);
    var _layer = this._getLayer(id);
    if (_layer) {
      delete this.layers[this.layers.indexOf(_layer)]; // `this.layers` should be replaced with `this._layers`
    }
    this._update();
    return this;
  }
```

Instead of `this.layers` should be `this._layers` i guess.

**Comment by ismyrnow** [comment]
You're totally right about that. Would someone mind opening a pull request with the fix?

**Comment by ttsvetko** [comment]
I can do later

**Comment by ForgottenLords** [comment]
calling delete on the array index is not sufficient, as later it is possible to iterate over the array and delete leaves the value at that index 'undefined'.  Instead, you must splice the item from the array to leave it continuously defined.

I had included a fix in my pull request #46 (an unrelated feature request)

**Comment by karlitos** [comment]
Please, accept the PR to fix the issue caused by using _delete_ on array.

**Comment by newmanw** [comment]
Possible to merge this pull request and release a new version to npm?
